src/bilder/images/edxapp_v2/deploy.py

```diff
@@ -0,0 +1,368 @@
+    ConsulTemplate,
+    ConsulTemplateConfig,
+    ConsulTemplateTemplate,
+    ConsulTemplateVaultConfig,
+)
+from bilder.components.hashicorp.consul_template.steps import (
+    consul_template_permissions,
+)
+from bilder.components.hashicorp.steps import (
+    configure_hashicorp_product,
+    install_hashicorp_products,
+    register_services,
+)
+from bilder.components.hashicorp.vault.models import (
+    Vault,
+    VaultAgentCache,
+    VaultAgentConfig,
+    VaultAutoAuthAWS,
+    VaultAutoAuthConfig,
+    VaultAutoAuthMethod,
+    VaultAutoAuthSink,
+    VaultAutoAuthFileSink,
+    VaultConnectionConfig,
+    VaultListener,
+    VaultTCPListener,
+)
+from bilder.facts.has_systemd import HasSystemd
+from bilder.lib.linux_helpers import DOCKER_COMPOSE_DIRECTORY
+from bilder.lib.template_helpers import (
+    CONSUL_TEMPLATE_DIRECTORY,
+    place_consul_template_file,
+)
+from bridge.lib.magic_numbers import VAULT_HTTP_PORT
+from bridge.lib.versions import (
+    CONSUL_TEMPLATE_VERSION,
+    CONSUL_VERSION,
+    TUTOR_PERMISSIONS_VERSION,
+    VAULT_VERSION,
+)
+from bridge.secrets.sops import set_env_secrets
+
+from bilder.images.edxapp_v2.lib import OPENEDX_RELEASE, WEB_NODE_TYPE, node_type
+from bridge.settings.openedx.accessors import fetch_application_version
+from bridge.settings.openedx.types import OpenEdxApplication
+
+VERSIONS = {
+    "consul": os.environ.get("CONSUL_VERSION", CONSUL_VERSION),
+    "consul-template": os.environ.get(
+        "CONSUL_TEMPLATE_VERSION", CONSUL_TEMPLATE_VERSION
+    ),
+    "tutor_permissions": os.environ.get(
+        "TUTOR_PERMISSIONS_VERSION", TUTOR_PERMISSIONS_VERSION
+    ),
+    "vault": os.environ.get("VAULT_VERSION", VAULT_VERSION),
+}
+TEMPLATES_DIRECTORY = Path(__file__).resolve().parent.joinpath("templates")
+FILES_DIRECTORY = Path(__file__).resolve().parent.joinpath("files")
+
+set_env_secrets(Path("consul/consul.env"))
+
+# TODO MD 20230515 Probably don't need this anymore.
+apt.packages(
+    name="Remove unattended-upgrades to prevent race conditions during build.",
+    packages=["unattended-upgrades"],
+    present=False,
+)
+
+# Setup some environment variables that will be pulled in by docker / docker-compose
+EDX_INSTALLATION_NAME = os.environ.get("EDX_INSTALLATION", "mitxonline")
+edx_version = fetch_application_version(
+    OPENEDX_RELEASE, EDX_INSTALLATION_NAME, OpenEdxApplication.edxapp
+)
+files.put(
+    name=f"Setting the EDXAPP_DOCKER_TAG env var to {OPENEDX_RELEASE}-{EDX_INSTALLATION_NAME}",  # noqa: E501
+    src=io.StringIO(f"{OPENEDX_RELEASE}-{EDX_INSTALLATION_NAME}"),
+    dest="/etc/default/edxapp_docker_tag",
+)
+files.put(
+    name=f"Settings the TUTOR_PERMISSIONS_VERSION env var to {VERSIONS['tutor_permissions']}",  # noqa: E501
+    src=io.StringIO(VERSIONS["tutor_permissions"]),
+    dest="/etc/default/tutor_permissions_tag",
+)
+
+# Preload some docker images. This will accelerate the first startup
```

Comment:
The docker image is 5.87 GB extracted and this step adds a non-trivial amount of time to the AMI build which means it is saving a substantial amount of startup time. Worth evaluating in elsewhere if we've got other big docker images in use. 

This 5.87GB also explodes out the root FS on the AMI which is mentioned and addressed in `custom_install.pkr.hcl`